libavcodec/qsvenc: Let runtime to set default parameter:

wenbinc-Bin opened the pull request:
Unset qsv_h264 and qsv_hevc's default settings. Let runtime to decide this
parameter, so that vpl can choose the best parameter and ffmpeg-qsv can
keep up with runtime's update.

Signed-off-by: Wenbin Chen <wenbin.chen@intel.com>

wenbinc-Bin committed:
libavcodec/qsvenc: Let runtime to set default parameter.

Unset qsv_h264 and qsv_hevc's default settings. Let runtime to decide
these parameters, so that it can choose the best parameter and ffmpeg-qsv
can keep up with runtime's update.

Signed-off-by: Wenbin Chen <wenbin.chen@intel.com>

wenbinc-Bin commented:
@xhaihao I change the default setting as what we discussed with Chunbo. Can you help to review?

xhaihao commented:
Could you please replace 'vpl runtime' with 'runtime' ? it works for mediasdk runtime too.

wenbinc-Bin commented:
Wait, setting targetKbps to 0 on h264 makes MSDK report error.

wenbinc-Bin commented:
> Could you please replace 'vpl runtime' with 'runtime' ? it works for mediasdk runtime too.

Ok

wenbinc-Bin commented:
I update the PR. For h264_qsv problem, there are zero check in runtime [https://github.com/oneapi-src/oneVPL-intel-gpu/blob/175cd3ee088e09eed62e94ec6f34381b683d1df4/_studio/mfx_lib/encode_hw/h264/src/mfx_h264_enc_common_hw.cpp#L2038](https://github.com/oneapi-src/oneVPL-intel-gpu/blob/175cd3ee088e09eed62e94ec6f34381b683d1df4/_studio/mfx_lib/encode_hw/h264/src/mfx_h264_enc_common_hw.cpp#L2038). I need to ask Chunbo.

wenbinc-Bin committed:
libavcodec/qsvenc: Let runtime to decide targetUsage

Set preset default value to MFX_TARGETUSAGE_UNKNOWN. Let runtime to
decide the targetUsage, so that ffmpeg-qsv can keep up with runtime's
update.

Signed-off-by: Wenbin Chen <wenbin.chen@intel.com>

wenbinc-Bin commented:
@xhaihao I update the PR. I don't change async_depth to default, because ffmpeg-qsv uses it to create a pool. Only change async_depth to default won't work, and it need other changes.

wenbinc-Bin commented:
@xhaihao This PR is ready for review. Could you please help to review it? Thanks.

wenbinc-Bin commented:
submit to community: [https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=7614](https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=7614)
